api/envoy/config/listener/v3/listener_components.proto

```diff
@@ -109,6 +109,113 @@ message FilterChainMatch {
     EXTERNAL = 2;
   }
 
+  // Matches filter chains by the destination port.
+  message DestinationPort {
+    // Optional destination port to consider when use_original_dst is set on the
+    // listener in determining a filter chain match.
+    google.protobuf.UInt32Value destination_port = 1
```

Comment:
If we are going to make this more generic anyway, can we change this to a range or list similar to source ports?

Reply:
Ack, makes sense.

Reply:
If we have a range, do we need the single port? You can always put a single port into the range definition so there is no backwards compat issue?

Reply:
It looks rather ugly to have one element range for many chains. Majority of listeners with original_dst just want one port.